controllers/setupphase_controller.go

```diff
@@ -0,0 +1,161 @@
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	v1alpha1 "github.com/suskin/stack-template-engine/api/v1alpha1"
+)
+
+// StackConfigurationReconciler reconciles a StackConfiguration object
+type SetupPhaseReconciler struct {
+	Client  client.Client
+	Log     logr.Logger
+	Manager manager.Manager
+}
+
+type Behavior struct {
+	cfg *v1alpha1.StackConfigurationBehavior
+	gvk *schema.GroupVersionKind
+}
+
+const (
+	setupTimeout = 60 * time.Second
+)
+
+// +kubebuilder:rbac:groups=helm.samples.stacks.crossplane.io,resources=stackconfigurations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=helm.samples.stacks.crossplane.io,resources=stackconfigurations/status,verbs=get;update;patch
+
+func (r *SetupPhaseReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), setupTimeout)
+	defer cancel()
+
+	i := &v1alpha1.StackConfiguration{}
+	if err := r.Client.Get(ctx, req.NamespacedName, i); err != nil {
+		if kerrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	r.Log.V(0).Info("Hello World!", "instanceName", req.NamespacedName, "instance", i)
+
+	return ctrl.Result{}, r.setup(i)
+}
+
+func (r *SetupPhaseReconciler) setup(sc *v1alpha1.StackConfiguration) error {
+	// For each behavior:
+	// - Grab the configuration values:
+	//   * Source stack; image or url
+	//   * Source GVK
+	//   * Event type
+	//   * The coordinates of the behavior in the configuration, so it can be found again. Or,
+	//     the behavior itself
+	// - Create a render controller, passing it the configuration values
+
+	// Questions
+	// Should we grab all of the configuration at setup time, or at render time?
+	// - At render time, so that we're always using the latest version of the object
+	// - Though, the ideal would be if we cached the configuration and changed it if it changed
+
+	behaviors := r.getBehaviors(sc)
+
+	for _, b := range behaviors {
+		gvk := b.gvk
+		// TODO we don't want to be hard-coding the event name here.
+		event := v1alpha1.EventName("reconcile")
+
+		if err := r.NewRenderController(gvk, event); err != nil {
+			// TODO what do we want to do if some of the registrations succeed and some of them fail?
+			r.Log.Error(err, "Error creating new render controller!", "gvk", gvk)
+		}
+	}
+
+	return nil
+}
+
+// This exists because getting the individual behaviors may be a bit tricker in the future.
+// For example, the engine may be configured at multiple levels. Another example is that
+// behaviors may be configured at multiple levels, if there are stack-level behaviors in
+// addition to object-level behaviors.
+func (r *SetupPhaseReconciler) getBehaviors(sc *v1alpha1.StackConfiguration) []Behavior {
+	scbs := sc.Spec.Behaviors.CRDs
+
+	behaviors := make([]Behavior, 0)
+
+	for rawGvk, scb := range scbs {
+		// We are assuming strings look like "Kind.group.com/version"
+		gvkSplit := strings.SplitN(string(rawGvk), ".", 2)
```

Comment:
Yeah. I've been considering this but haven't made any changes yet. The reason it's a single field currently is so that we can use the GVK as a key in the stack behaviors configuration, which seems a little nicer for users to read and write:

```
behaviors:
  crds:
    Kind.group/version:
      # my configuration
```

The alternative would look more like:

```
behaviors:
  crds:
  - apiVersion: group/version
    kind: Kind
    # my configuration would be other keys in this object
```

We might still go that direction though. I've got in on a list of things to revisit after we have the first working system, haha